torch/_refs/__init__.py

```diff
@@ -3738,9 +3738,8 @@ def roll(
     dim = dims[0]
     size = a.shape[dim]
     start = (size - shifts[0]) % size
-    t0 = torch.narrow(a, dim, start, size - start)
-    t1 = torch.narrow(a, dim, 0, start)
-    return torch.cat((t0, t1), dim)
+    idx = torch.arange(size, device=a.device)
+    return a.index_select(dim, torch.fmod(start + idx, size))
```

Comment:
Nice. I take it the codegen is the same?

Reply:
I tested on the OpInfos running it with `PYTORCH_ALL_SAMPLES=1 TORCH_LOGS_FORMAT="%(message)s" TORCH_LOGS=output_code pytest test/inductor/test_torchinductor_opinfo.py -vsk test_comprehensive_roll_cuda_int64` and it does.